doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+    
+            {
+                "status": 200,
+                "username": "the_username",
+                "rank": 1
+            }
+            
+    + Schema
+    
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "username": {
+                        "description": "The authenticated users name",
+                        "type": "string"
+                    },
+                    "rank": {
+                        "description": "The authenticated users rank",
+                        "type": "integer"
+                    }
+                },
+                "required": [
+                    "status",
+                    "username",
+                    "rank"
+                ],
+                "additionalProperties": false
+            }
+
++ Response 401 (application/json)
+    + Body
+    
+            {
+                "status": 401,
+                "error": "Unauthorized",
+                "message": "You need to be authenticated to perform this action.",
+                "loca": "NEED_AUTHENTICATION"
+            }
+            
+    + Schema
+
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "error": {
+                        "description": "Textual description of HTTP status code",
+                        "type": "string"
+                    },
+                    "message": {
+                        "description": "Detailed information about the error.",
+                        "type": "string"
+                    },
+                    "loca": {
+                        "description": "Unique message token that can be used for localization in a frontend application.",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "status",
+                    "message",
+                    "loca"
+                ],
+                "additionalProperties": false
+            }
+        
+
+## Register [POST /register]
+
+This method can be used to register for the service. An user account will grant access to more functionality.
+
++ Request Register via form (application/x-www-form-urlencoded)
+    + Attributes
+        + username: the_username (string, required)
+        
+            The username must be between 3 and 20 signs long and may contain only letters a-z (lower- and upper-case), the numbers 0-9 and dots (.) as well as dashes (-).
+            
+        + password: somePassword (string, required) - A non-empty password
```

Comment:
Currently not. I always find it very difficult to establish useful rules for passwords, because in many occasions I'm annoyed by too restrictive rules. But I'm open for suggestions!